tests/faucet_mininet_test_unit.py

```diff
@@ -3297,3 +3297,509 @@ def test_switching(self):
+
+    def ping_between_hosts(self, users):
+        """Ping between the specified host
+        Args:
+            users (list<mininet.host>): users to ping between.
+                0 & 1 should be authenitcated.
+                2 should be unauthenticated,
+        """
+        for user in users:
+            user.defaultIntf().updateIP()
+
+        h0 = users[0]
+        h1 = users[1]
+        h2 = users[2]
+        h1_ip = ipaddress.ip_address(unicode(h1.IP()))
+        # h2 will not have an ip via dhcp as they are unauthenticated, so give them one.
+        h2.setIP('10.0.12.253')
+        h2_ip = ipaddress.ip_address(unicode(h2.IP()))
+        # ping between the authenticated hosts
+        self.one_ipv4_ping(h0, h1_ip)
+        self.one_ipv4_ping(h1, '10.0.0.2')
+
+        #ping between an authenticated host and an unauthenticated host
+        try:
+            self.one_ipv4_ping(h1, h2_ip)
+        except AssertionError:
+            pass
+        else:
+            self.fail('{0} {1} should not be able to ping {2} {3}'.format(h1, h1.IP(), h2, h2.IP()))
+ 
+        ploss = self.net.ping(hosts=[users[0], users[2]], timeout='5')
+        self.assertAlmostEqual(ploss, 100)
+        try:
+            self.one_ipv4_ping(h0, h2_ip)
+        except AssertionError:
+            pass
+        else:
+            self.fail('{0} {1} should not be able to ping {2} {3}'.format(h0, h0.IP(), h2, h2.IP()))
+
+    def test_onlydot1x(self):
+        """Only authenticate through dot1x"""
+        users = self.clients
+
+        self.logon_dot1x(users[0])
+        self.logon_dot1x(users[1])
+        time.sleep(5)
+        self.ping_between_hosts(users)
+
+
+class FaucetSingleAuthenticationNoLogOnTest(FaucetAuthenticationSingleSwitchTest):
+    """Check the connectivity when the hosts are not authenticated"""
+
+    def test_nologon(self):
+        """Get the users to ping each other before anyone has authenticated
+        """
+        users = self.clients
+        for user in users:
+            cmd = "ip addr flush {0}-eth0 && dhcpcd --timeout 5 {0}-eth0".format(
+                user.name)
+            user.cmd(cmd)
+            user.defaultIntf().updateIP()
+
+        ploss = self.net.ping(hosts=users, timeout='5')
+        self.assertAlmostEqual(ploss, 100)
+
+
+class FaucetSingleAuthenticationDot1XLogonAndLogoffTest(FaucetAuthenticationSingleSwitchTest):
+    """Log on using dot1x and log off"""
+
+    def test_logoff(self):
+        """Check that the user cannot go on the internet after logoff"""
+        h0 = self.clients[0]
+        interweb = self.net.hosts[1]
+
+        self.logon_dot1x(h0)
+        self.one_ipv4_ping(h0, interweb.IP())
+        result = self.check_http_connection(h0)
+        self.assertTrue(result)
+
+        h0.cmd("wpa_cli logoff")
+        # TODO possibly poll wpa_cli status to check that the status has changed?
```

Comment:
The problem with polling 'wpa_cli status' is that the status will change regardless of whether the 'switch' has received the logoff or not, and the client will be unaware. So in theory the status could be updated before the logoff actions have occurred.